src/components/structures/RightPanel.tsx

```diff
@@ -348,7 +355,8 @@ export default class RightPanel extends React.Component<IProps, IState> {
                     initialEvent={this.state.initialEvent}
                     initialEventHighlighted={this.state.initialEventHighlighted}
                     permalinkCreator={this.props.permalinkCreator}
-                    e2eStatus={this.props.e2eStatus} />;
+                    e2eStatus={this.props.e2eStatus}
+                    previousPhase={RightPanelStore.getSharedInstance().previousPhase} />;
```

Comment:
Hmm, it feels a bit odd to pass the store's value as a prop like this... Perhaps `ThreadView` can ask the store for this value internally?

Reply:
This is a good idea. In the beginning I wanted to keep the current bahaviour of the back button of the TreadView always going back to the threadPanel except if the thread view was opened from the timlelineCard. This is why I also had the previousPhase property in the EventTile. I would have needed to know where the "thread open request" originated from, to decided what the back button does. But since I new went with this simpler approach it is possible to just ask for the previousPhase in the threads view.
This however does change the behaviour!
RoomSummary is open -> click on a thread in the timeline -> click the back button in the ThreadView -> the RoomSummary will open (I think) before this opend the threadPanel

Reply:
I now fixed it with some additional logic to only allow the back button to bring you to the threads panel or the timeline.
See: [`d6ad1ae` (#7112)](https://github.com/matrix-org/matrix-react-sdk/pull/7112/commits/d6ad1aeb3439e293684cd863cbff7b9dfb0bb96f)